### PR TITLE
fix: Fix the potential out-of-bounds vulnerability in the slice withi…

### PR DESCRIPTION
**Pull request by sologgfun**
Hello Team,
I've been experimenting with this project and encountered an error that I believe is a bug.
image
After investigating and modifying the source code, I've managed to fix the issue and have tested it locally. However, I'm not sure how to reproduce the scenario that triggers this bug.

### CODE REVIEW

File: agent/protocol/mysql/handler.go [hunk 1]
```diff
@@ -425,11 +425,15 @@ func HandleStmtCloseRequest(reqPacket *MysqlPacket, prepareMap map[int]PreparedS
...
 	record.Req.(*MysqlPacket).msg = ""
-
-	stmt_id, _ :=
-		common.LEndianBytesToKInt[int32]([]byte(reqPacket.msg[kStmtIDStartOffset:]), kStmtIDBytes)
+	var stmt_id int32
+	if len(reqPacket.msg) >= kStmtIDStartOffset+kStmtIDBytes {
+		stmt_id, _ = common.LEndianBytesToKInt[int32]([]byte(reqPacket.msg[kStmtIDStartOffset:]), kStmtIDBytes)
+	} else {
+		// 处理错误情况，例如记录日志或返回错误
+		common.ProtocolParserLog.Errorf("reqPacket.msg 长度不足")
```
Comment (hengyoush) [inline agent/protocol/mysql/handler.go:435]: log should be english

**Comment by hengyoush** [comment]
Thank you for your contribution, i leave a comment, pls take a look.

File: agent/protocol/mysql/handler.go (hunk 1 above)
Comment (sologgfun) [reply agent/protocol/mysql/handler.go]: I agree, I will fix it later

**Comment by sologgfun** [comment]
> Thank you for your contribution, i leave a comment, pls take a look.

I commit a new one , do i need to squash the 2 commit  in 1?

**Comment by hengyoush** [comment]
> > Thank you for your contribution, i leave a comment, pls take a look.
> 
> I commit a new one , do i need to squash the 2 commit in 1?

No need, i will squash it when merge this pr.

**Review by hengyoush — APPROVED** [review]
LGTM

**Pull request merged**